recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useRef, useState } from 'react';
+import DropdownButton from './DropdownButton/DropdownButton';
+import DropdownContent from './DropdownContent/DropdownContent';
+
+
+type DropdownProps = {
+    buttonText: string;
+    content?: React.ReactNode;
+  };
+
+const Dropdown: React.FC<DropdownProps> = ({buttonText,content}) => {
+
+    const [open, setOpen] = useState(false);
+
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    const handleClickOutsideDropdown =(e:any)=> {
+        if(open && !dropdownRef.current?.contains(e.target as Node)){
+            setOpen(false);
+        }
+    }
+    window.addEventListener("click",handleClickOutsideDropdown);
```

Comment:
Nice touch, but it would be better to wrap this into an useEffect along with the handleClickOutsideOpen function in order to avoid the browser adding the event listener at every render. Set the dependencies for the useEffect to be `dropdownRef` and `open`.
Also, in the useEffect function, return a function that removes the event listener you're setting.
It should look something like this:
![image](https://github.com/user-attachments/assets/afd83f12-1ab8-430b-99f3-d293e61c57f8)